src/handler/grpc/request/event.rs

```diff
@@ -118,4 +153,160 @@ impl Event for Eventer {
+            filename,
+            offset,
+            chunk_size
+        );
+
+        let chunk = match infra::cache::file_data::disk::get(
+            path,
+            Some(Range {
+                start: offset,
+                end: offset + chunk_size,
+            }),
+        )
+        .await
+        {
+            Some(file_data) => file_data,
+            None => {
+                if let Err(e) = tx.send(Err(Status::not_found(path))).await {
+                    log::error!("Failed to send error: {}", e);
+                }
+                return Err(Status::not_found(path));
+            }
+        };
+
+        let response = FileContentResponse {
+            entries: vec![FileContent {
+                content: chunk.to_vec(),
+                filename: filename.clone(),
+            }],
+        };
+
+        if let Err(e) = tx.send(Ok(response)).await {
+            log::error!("Failed to send file chunk: {}", e);
+            return Err(Status::internal("Failed to send file chunk"));
+        }
+
+        offset += chunk_size;
+    }
+
+    log::info!(
+        "handle file:{}, total_size: {}, offset: {} elapsed: {}ms",
+        path,
+        total_size,
+        offset,
+        start.elapsed().as_millis()
+    );
+
+    Ok(())
+}
+
+async fn get_files_from_notifier(addr: &str, filekey: &str) -> Result<()> {
+    if !get_config().cache_latest_files.download_from_node_enabled {
+        return Err(anyhow::anyhow!(DOWNLOAD_FROM_NODE_NOT_ENABLED));
+    }
+
+    let start = std::time::Instant::now();
+    log::debug!("get_files_from_notifier start, file: {}", filekey);
+    let token: MetadataValue<_> = get_internal_grpc_token()
+        .parse()
+        .map_err(|_| anyhow::anyhow!("Invalid token"))?;
+
+    let channel = crate::service::grpc::get_cached_channel(addr).await?;
+    let client = EventClient::with_interceptor(channel, move |mut req: Request<()>| {
+        req.metadata_mut().insert("authorization", token.clone());
+        Ok(req)
+    });
+
+    let mut request = Request::new(SimpleFileList {
+        paths: vec![filekey.to_string()],
+    });
+    request.set_timeout(std::time::Duration::from_secs(
+        get_config().limit.query_timeout,
+    ));
+
+    let response = client
+        .send_compressed(CompressionEncoding::Gzip)
+        .accept_compressed(CompressionEncoding::Gzip)
+        .max_decoding_message_size(250 * 1024 * 1024)
+        .max_encoding_message_size(250 * 1024 * 1024)
+        .get_files(request)
+        .await
+        .map_err(|e| anyhow::anyhow!("Failed to get file: {filekey} from {addr}, {e}"))?;
+
+    let mut response_stream = response.into_inner();
+    let mut file_content = Vec::new();
+
+    while let Some(response) = response_stream.next().await {
+        let response =
+            response.map_err(|e| anyhow::anyhow!("Failed to receive file chunk: {}", e))?;
+        if let Some(content) = response.entries.first() {
```

Comment:
the server side is accept multiple files, but response only get the first entry?